hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/ApproveAllowanceChecks.java

```diff
@@ -20,34 +20,264 @@
+		if (validity != OK) {
+			return validity;
+		}
+
+		validity = validateNftAllowances(nftAllowances, payerAccount, accountStore, tokenStore);
+		if (validity != OK) {
+			return validity;
+		}
+
+		return OK;
+	}
+
+	/**
+	 * Validates the CryptoAllowances given in {@link com.hederahashgraph.api.proto.java.CryptoApproveAllowance}
+	 * transaction
+	 *
+	 * @param cryptoAllowances
+	 * 		crypto allowances list
+	 * @param payerAccount
+	 * 		Account of the payer for the approveAllowance txn
+	 * @param accountStore
+	 * 		account store
+	 * @return response code after validation
+	 */
+	ResponseCodeEnum validateCryptoAllowances(
+			final List<CryptoAllowance> cryptoAllowances,
+			final Account payerAccount,
+			final AccountStore accountStore) {
+		if (cryptoAllowances.isEmpty()) {
+			return OK;
+		}
+
+		for (final var allowance : cryptoAllowances) {
+			final var owner = Id.fromGrpcAccount(allowance.getOwner());
+			final var spender = Id.fromGrpcAccount(allowance.getSpender());
+
+			final var fetchResult = fetchOwnerAccount(owner, payerAccount, accountStore);
+			if (fetchResult.getRight() != OK) {
+				return fetchResult.getRight();
+			}
+			final var ownerAccount = fetchResult.getLeft();
+
+			var validity = validateAmount(allowance.getAmount());
+			if (validity != OK) {
+				return validity;
+			}
+			validity = validateSpender(ownerAccount.getId(), spender);
+			if (validity != OK) {
+				return validity;
+			}
+		}
+		return OK;
+	}
+
+	/**
+	 * Validate fungible token allowances list {@link com.hederahashgraph.api.proto.java.CryptoApproveAllowance}
+	 * transaction
+	 *
+	 * @param tokenAllowances
+	 * 		token allowances list
+	 * @param payerAccount
+	 * 		Account of the payer for the approveAllowance txn
+	 * @param accountStore
+	 * 		account store
+	 * @param tokenStore
+	 * 		read only token store
+	 * @return
+	 */
+	ResponseCodeEnum validateFungibleTokenAllowances(
+			final List<TokenAllowance> tokenAllowances,
+			final Account payerAccount,
+			final AccountStore accountStore,
+			final ReadOnlyTokenStore tokenStore) {
+		if (tokenAllowances.isEmpty()) {
+			return OK;
+		}
+
+		for (final var allowance : tokenAllowances) {
+			final var owner = Id.fromGrpcAccount(allowance.getOwner());
+			final var spender = Id.fromGrpcAccount(allowance.getSpender());
+			final var token = tokenStore.loadPossiblyPausedToken(Id.fromGrpcToken(allowance.getTokenId()));
+
+			final var fetchResult = fetchOwnerAccount(owner, payerAccount, accountStore);
+			if (fetchResult.getRight() != OK) {
+				return fetchResult.getRight();
+			}
+			final var ownerAccount = fetchResult.getLeft();
+			if (!token.isFungibleCommon()) {
+				return NFT_IN_FUNGIBLE_TOKEN_ALLOWANCES;
+			}
+
+			var validity = validateTokenAmount(allowance.getAmount(), token);
+			if (validity != OK) {
+				return validity;
+			}
+
+			validity = validateTokenBasics(ownerAccount, spender, token, tokenStore);
```

Comment:
nit: we can directly return here instead of the OK check.

Reply:
This is inside a for loop , if we return it doesn't go through the next allowance